tests/python/gaia-ui-tests/gaiatest/gaia_graphics_test.py

```diff
@@ -48,17 +50,20 @@ def tearDown(self):
         """At the end of test execution, it checks for the errors"""
         self.assertTrue(self.test_passed, msg=self.failcomment)
 
-    def take_screenshot(self):
-        """invokes screen capture event, crops the status bar, and saves to the file"""
+    # hook to override default locale setting
+    # see https://github.com/mozilla-b2g/gaia/blob/master/tests/python/gaia-ui-tests/gaiatest/gaia_test.py#L1019
+    def modify_settings(self, settings):
+        settings['language.current'] = self.locale
```

Comment:
Only trick with this is now it'll be significant whether you call modify_settings first or last (one clobbers you w/ its locale, the other does it the other way around). That needs to be communicated somehow.

However, this really misses the point of this method. It's a user hook for a -test- to modify settings, not somewhere for the harness to modify its own settings. It's basically a virtual abstract method.

My fairly strong suggestion is to -not- put the locale override here, instead do it when initial settings are set up (initialize that dict from cmd line instead of just assigning 'en-US') and keep modify_settings() as purely to do per-test overrides.